### PR TITLE
Hotfix: Fix offset calculation to prevent overflow if offset is really large

### PR DESCRIPTION
**Pull request by amcamd**
- Contains:
  - Torre's fix to add offset to array pointers in call to RocblasContractionProblem
  - Andrew's trsm changes to avoid int32_t overflow in trsm offset calculations
  - Alex's tensile_tag.txt for changes in Tensile

### CODE REVIEW

**Review by TorreZuk — APPROVED** [review]
All looks good to me, and same changes as passing in develop.  Only thing to watch is if how the Tensile hotfix get's merged first as may break the required tensile_tag.

**Comment by TorreZuk** [comment]
@amcamd not sure how you have clang-format failure if you cherry-picked

**Comment by amcamd** [comment]
I had to resolve a merge conflict, from ROCm 5.5 to now trtri_gemm_block has changed to rocblas_trtri_gemm_block. I needed to reverse the change. I guess that clang-format did not run on the commit after resolving the merge conflict. I will fix it.

File: clients/gtest/trsm_gtest.yaml [hunk 1]
```diff
@@ -60,6 +60,15 @@ Definitions:
   - &large_memory_matrix_size_range
     - { M: 8320, N: 128, lda: 8320, ldb: 8320 }
 
+  - &size_t_left_matrix_size_range
+#   - { M: 4, N: 46435, lda: 4, ldb: 46435 }
+    - { M: 46345, N: 4, lda: 46345, ldb: 46345 }
+#   - { M: 47000, N: 4, lda: 47000, ldb: 47000 } # calls rocblas_internal_gemm_template with batch_count=367, stride_a=6016128
+
+  - &size_t_right_matrix_size_range
+    - { M: 4, N: 46345, lda: 46345, ldb: 4 }
```
Comment (daineAMD) [inline clients/gtest/trsm_gtest.yaml:69]: These tests timed out for me on gfx90a, but passed when I increased the timeout threshold. I'm testing all other configs as well just to be safe.

Comment (amcamd) [reply clients/gtest/trsm_gtest.yaml:69]: @daineAMD  Naveen's speedup for the initialization of the triangle matrix in https://github.com/ROCmSoftwarePlatform/rocBLAS-internal/pull/1551 . This PR is in the develop branch but it is not in release/rocm-rel-5.5. Tests will run faster on the develop branch than on release/rocm-rel-5.5.

**Pull request merged**